skiko/src/awtMain/cpp/linux/swingRedrawer.cc

```diff
@@ -0,0 +1,187 @@
+#include <jawt_md.h>
+#include <GL/gl.h>
+#include <GL/glx.h>
+#include <X11/X.h>
+#include <X11/Xlib.h>
+#include <X11/Xresource.h>
+#include <cstdlib>
+#include <unistd.h>
+#include <stdio.h>
+#include "jni_helpers.h"
+
+class OffScreenContext
+{
+public:
+    Display* display;
+    GLXContext context;
+    GLXFBConfig* fbConfigs;
+
+    OffScreenContext() {
+        const int glxContextAttribs[] {
+            GLX_DRAWABLE_TYPE, GLX_WINDOW_BIT,
+            GLX_RENDER_TYPE, GLX_RGBA_BIT,
+            GLX_DOUBLEBUFFER, False,
+            GLX_RED_SIZE, 8,
+            GLX_GREEN_SIZE, 8,
+            GLX_BLUE_SIZE, 8,
+            None
+        };
+
+        display = XOpenDisplay(nullptr);
+
+        int numConfigs = 0;
+        fbConfigs = glXChooseFBConfig(display, DefaultScreen(display), glxContextAttribs, &numConfigs);
+
+        XVisualInfo* visual = glXGetVisualFromFBConfig(display, fbConfigs[0]);
+
+        context = glXCreateContext(display, visual, nullptr, True);
```

Comment:
Should we do sanity check during initialising so we could fall back to software render instead of crashing (or UB of some sort)? 
[XOpenDisplay](https://www.x.org/releases/X11R7.5/doc/man/man3/XOpenDisplay.3.html), [glXChooseFBConfig](https://registry.khronos.org/OpenGL-Refpages/gl2.1/xhtml/glXChooseFBConfig.xml), [glXGetVisualFromFBConfig](https://registry.khronos.org/OpenGL-Refpages/gl2.1/xhtml/glXGetVisualFromFBConfig.xml), [glXCreateContext](https://registry.khronos.org/OpenGL-Refpages/gl2.1/xhtml/glXCreateContext.xml) all can return null.